modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java

```diff
@@ -2424,6 +2432,112 @@ public void testExecuteObjectActionWithUsePreferredLanguageForGuestsParameter()
+			_objectActionExecutorRegistry.getObjectActionExecutor(
+				0, ObjectActionExecutorConstants.KEY_GROOVY),
+			"_objectScriptingExecutor", _originalObjectScriptingExecutor);
+
+		try (Closeable closeable =
+				ScriptManagementConfigurationTestUtil.saveWithCloseable(true)) {
+
+			ObjectDefinition objectDefinition = null;
+
+			try {
+				objectDefinition =
+					ObjectDefinitionTestUtil.publishObjectDefinition(
+						Collections.singletonList(
+							new TextObjectFieldBuilder(
+							).labelMap(
+								LocalizedMapUtil.getLocalizedMap("First Name")
+							).name(
+								"firstName"
+							).required(
+								true
+							).build()),
+						ObjectDefinitionConstants.SCOPE_COMPANY);
+
+				Role role = _roleLocalService.getRole(
+					TestPropsValues.getCompanyId(), RoleConstants.GUEST);
+
+				ResourceActionLocalServiceUtil.addResourceAction(
+					objectDefinition.getResourceName(), ActionKeys.VIEW, 1);
+
+				_resourcePermissionLocalService.addResourcePermission(
+					TestPropsValues.getCompanyId(),
+					objectDefinition.getResourceName(),
+					ResourceConstants.SCOPE_COMPANY,
+					String.valueOf(TestPropsValues.getCompanyId()),
+					role.getRoleId(), "ADD_OBJECT_ENTRY");
+				_resourcePermissionLocalService.addResourcePermission(
+					TestPropsValues.getCompanyId(),
+					objectDefinition.getClassName(),
+					ResourceConstants.SCOPE_COMPANY,
+					String.valueOf(TestPropsValues.getCompanyId()),
+					role.getRoleId(), ActionKeys.VIEW);
+
+				Class<?> clazz = ObjectActionLocalServiceTest.class;
+
+				_addObjectAction(
+					objectDefinition.getObjectDefinitionId(),
+					ObjectActionExecutorConstants.KEY_GROOVY,
+					ObjectActionTriggerConstants.KEY_ON_AFTER_ADD,
+					UnicodePropertiesBuilder.create(
+						true
+					).put(
+						"parameters",
+						JSONUtil.put(
+							"objectName", objectDefinition.getShortName()
+						).toString()
+					).put(
+						"script",
+						StringUtil.read(
+							clazz,
+							StringBundler.concat(
+								"/com/liferay/object/service/test/dependencies",
+								"/", clazz.getSimpleName(), ".",
+								testName.getMethodName(), ".groovy"))
+					).build());
+
+				try (LogCapture logCapture =
+						LoggerTestUtil.configureLog4JLogger(
+							"GROOVY_ACTION", LoggerTestUtil.INFO)) {
+
+					ObjectEntryTestUtil.addObjectEntry(
+						0L, objectDefinition.getObjectDefinitionId(),
+						Collections.singletonMap("firstName", "Peter"));
+
+					List<LogEntry> logEntries = logCapture.getLogEntries();
+
+					boolean scriptExecutado = logEntries.stream(
+					).anyMatch(
+						entry -> entry.getMessage(
+						).contains(
+							"Successful"
+						)
+					);
+
+					Assert.assertTrue("Did not run.", scriptExecutado);
+				}
+			}
+			finally {
+				ReflectionTestUtil.setFieldValue(
+					_objectActionExecutorRegistry.getObjectActionExecutor(
+						0, ObjectActionExecutorConstants.KEY_GROOVY),
+					"_objectScriptingExecutor",
+					_originalObjectScriptingExecutor);
```

Comment:
@Aquiiiles this is not needed. It's the same code from [here](https://github.com/liferay-bpm/liferay-portal/blob/bc1584fb291964d260bfa45711561d3bb06a3cc5/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java#L2439-L2442).